client/dashboard/customizable.js

```diff
@@ -70,7 +72,10 @@ const mergeSectionsWithDefaults = ( prefSections ) => {
 const CustomizableDashboard = ( { defaultDateRange, path, query } ) => {
 	const { updateUserPreferences, ...userPrefs } = useUserPreferences();
 
-	const sections = mergeSectionsWithDefaults( userPrefs.dashboard_sections );
+	const sections = useMemo(
+		() => mergeSectionsWithDefaults( userPrefs.dashboard_sections ),
+		[ userPrefs.dashboard_sections ]
+	);
```

Comment:
Make use of `useMemo` given the new object reference would be a little more expensive.